src/main/java/emissary/core/channels/LoggingInputStreamFactory.java

```diff
@@ -0,0 +1,199 @@
+package emissary.core.channels;
+
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * This class creates a InputStreamFactory for logging accesses to every method on the InputStream. It is intended only
+ * for development/debugging purposes.
+ */
+public final class LoggingInputStreamFactory {
```

Comment:
Suggest adding a `@Deprecated` annotation since we don't expect anyone to use this in committed code.

Reply:
I think this would be more confusing since the deprecated annotation indicates that the functionality will eventually be deleted and that is not the intention here.